typing/typecore.ml

```diff
@@ -4613,20 +4679,21 @@ and type_expect_
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
   | Pexp_while(scond, sbody) ->
-      let cond_env,wh_cond_region =
-        if is_local_returning_expr scond then env, false
-        else Env.add_region_lock env, true
+      let cond_env,wh_cond_region,mode=
```

Comment:
With the `if` gone, this code can now be tidied up. For instance `wh_cond_region` is no longer needed since it is always just `true`.